pilot/pkg/serviceregistry/kube/controller/ambient/waypoints.go

```diff
@@ -378,24 +378,35 @@ func (w *Waypoint) GetAddress() *workloadapi.GatewayAddress {
 	return w.Address
 }
 
+// if we find the HBONE listener, we always use it's WaypointSelector
+// if we find a PROXY listener, we may use the WaypointSelector for that listener when there is no HBONE listener
+// if we find neither, we default to the same namespace
 func makeAllowedRoutes(gateway *v1beta1.Gateway) WaypointSelector {
+	maybeWaypointSelector := WaypointSelector{
+		FromNamespaces: gatewayv1.NamespacesFromSame,
+	}
 	for _, l := range gateway.Spec.Listeners {
+		if l.Protocol == "istio.io/PROXY" {
```

Comment:
can we document this behavior somewhere? worht another name rather than all in capital letters.

Reply:
https://docs.google.com/document/d/1DAM_6SajCyC2hD8wBUraaZWZJqUVYPyemzw0-xpHmdY/edit?tab=t.0#bookmark=id.i2099vy9btwn is where the implementations come from.

That said, I think the implementation looking for that specific proto isn't actually what we want and I did another which bases the search on InboundBinding. It should be more correct.